libraries/AP_Common/Bitmask.h

```diff
@@ -44,29 +60,48 @@ class Bitmask {
 
     // clear all bits
     void clearall(void) {
-        for (uint16_t i=0; i<numwords; i++) {
-            bits[i] = 0;
-        }
+        memset(bits, 0, numwords);
     }
 
     // return true if given bitnumber is set
-    bool get(uint16_t bit) {
+    bool get(uint16_t bit) const {
         uint16_t word = bit/32;
         uint8_t ofs = bit & 0x1f;
         return (bits[word] & (1U << ofs)) != 0;
     }
 
     // return true if all bits are clear
-    bool empty(void) {
+    bool empty(void) const {
         for (uint16_t i=0; i<numwords; i++) {
             if (bits[i] != 0) {
                 return false;
             }
         }
         return true;
     }
-    
+
+    // return number of bits set
+    uint16_t count() const {
+        uint16_t sum = 0;
+        for (uint16_t i=0; i<numwords; i++) {
+            if (sizeof(bits[i]) <= sizeof(int)) {
```

Comment:
shouldn't this be int16_t?

Reply:
No, I think this is correct. `__builtin_popcount()` takes an `unsigned int`, `__builtin_popcountl()` an `unsigned long` and `__builtin_popcountll()` an `unsigned long long`.